src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,155 @@
+import classnames from 'classnames';
+import {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import { Props } from './TodoItem.props';
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  selectedTodos,
+  setSelectedTodos,
+  onUpdate,
+  selectedTodo,
+  setSelectedTodo,
+  todos,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [isClicked, setIsClicked] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const handleTitleUpdate = () => {
```

Comment:
useCallback 😭

Reply:
is it really useful to use hook useCallback without optimized child components? I watched few videos about it, and now I'm not sure about covering every function with it

Reply:
yeap. If you don’t use useCallback, your function will recreate each time when something in the component update